components/config/src/config/mod.rs

```diff
@@ -109,6 +109,7 @@ pub struct SerializedConfig<'a> {
     build_search_index: bool,
     extra: &'a HashMap<String, Toml>,
     markdown: &'a markup::Markdown,
+    search_index_format: &'a search::IndexFormat,
```

Comment:
You can create a new struct (have a look at how we do it for page and sections) for the data serialized so if we add more items it's nicely contained.